File: CodeGolf/DontPanic/DontPanic.js
z=readline;
[,,,t,y,,,u]=z().split(' ');e=[];[b,w,o,l]=["BLOCK","WAIT","RIGHT","LEFT"]
for(i=0;i<u; i++)a=z().split(' '),e[~~a[0]]=~~a[1]
while(1)[f,p,d]=z().split(' '),f=~~f,p=~~p,f<0||f==t?(g=y-p>0?o:l,r=g!=d?b:w):(q=e[f],h=q-p>0?o:l,r=q-p==0?w:h!=d?b:w),print(r)

File: CodeGolf/DontPanic/DontPanic.cpp
// Readable version

#include <iostream>
using namespace std;
int main()
{
    int n,w,r,x,p,t,a,e,f,o,c,l,v[15],u=0,i=0,s;
    cin>>n>>w>>r>>x>>p>>t>>a>>e;
    for(;i<e;i++)
    {
        cin>>f>>o;cin;
        v[f]=o;
    }
    while(1)
    {
        string d,m="WAIT";
        cin>>c>>l>>d;; 
        if(c==u){s=(x==c)?p:v[c];
        if((d=="RIGHT"&&l>s)||(d=="LEFT"&&l<s))m="BLOCK";u++;}
        cout<<m<<endl;
    }
}

// Code Golf

#include <iostream>
using namespace std;main(){int n,w,r,x,p,t,a,e,f,o,c,l,v[15],u=0,i=0,s;cin>>n>>w>>r>>x>>p>>t>>a>>e;for(;i<e;i++){cin>>f>>o;cin;v[f]=o;}while(1){string d,m="WAIT";cin>>c>>l>>d;;if(c==u){s=(x==c)?p:v[c];if((d=="RIGHT"&&l>s)||(d=="LEFT"&&l<s))m="BLOCK";u++;}cout<<m<<endl;}}